packages/controllers/src/snaps/SnapController.ts

```diff
@@ -69,104 +80,24 @@ import { Timer } from './Timer';
 
 export const controllerName = 'SnapController';
 
-export const SNAP_PREFIX_REGEX = new RegExp(`^${SNAP_PREFIX}`, 'u');
-
 export const SNAP_APPROVAL_UPDATE = 'wallet_updateSnap';
 
-type TruncatedSnapFields =
-  | 'id'
-  | 'initialPermissions'
-  | 'permissionName'
-  | 'version';
-
 const TRUNCATED_SNAP_PROPERTIES = new Set<TruncatedSnapFields>([
   'initialPermissions',
   'id',
   'permissionName',
   'version',
 ]);
 
-type RequestedSnapPermissions = {
-  [permission: string]: Record<string, Json>;
-};
-
-/**
- * A Snap as it exists in {@link SnapController} state.
- */
-export type Snap = {
-  /**
-   * Whether the Snap is enabled, which determines if it can be started.
-   */
-  enabled: boolean;
-
-  /**
-   * The ID of the Snap.
-   */
-  id: SnapId;
-
-  /**
-   * The initial permissions of the Snap, which will be requested when it is
-   * installed.
-   */
-  initialPermissions: RequestedSnapPermissions;
-
-  /**
-   * The Snap's manifest file.
-   */
-  manifest: SnapManifest;
-
-  /**
-   * Whether the Snap is blocked.
-   */
-  blocked: boolean;
-
-  /**
-   * Information detailing why the snap is blocked.
-   */
-  blockInformation?: BlockedSnapInfo;
-
-  /**
-   * The name of the permission used to invoke the Snap.
-   */
-  permissionName: string;
-
-  /**
-   * The source code of the Snap.
-   */
-  sourceCode: string;
-
-  /**
-   * The current status of the Snap, e.g. whether it's running or stopped.
-   */
-  status: Status;
-
-  /**
-   * The version of the Snap.
-   */
-  version: string;
-
-  /**
-   * The version history of the Snap.
-   * Can be used to derive when the Snap was installed, when it was updated to a certain version and who requested the change.
-   */
-  versionHistory: VersionHistory[];
-};
-
-export type VersionHistory = {
```

Comment:
Do we need to move all of this?

Reply:
Yeah, I tried moving as little as possible, but the required types use the other types, which use other types, and so on.